common/src/main/java/org/valkyrienskies/mod/mixin/server/world/MixinChunkMap.java

```diff
@@ -56,8 +58,10 @@ private void preReadChunk(final ChunkPos chunkPos, final CallbackInfoReturnable<
         final CompoundTag originalToReturn = compoundTag == null ? null :
             self.upgradeChunkTag(this.level.dimension(), this.overworldDataStorage, compoundTag, Optional.empty());
 
+        cir.setReturnValue(originalToReturn);
         if (originalToReturn == null) {
-            if (VSGameUtilsKt.isChunkInShipyard(level, chunkPos.x, chunkPos.z)) {
+            final ServerShip ship = VSGameUtilsKt.getShipManagingPos(level, chunkPos.x, chunkPos.z);
+            if (ship != null && !ShipSettingsKt.getSettings(ship).getShouldGenerateChunks()) {
```

Comment:
can we make this an `if ((ship == null && VSGameUtilsKt.isChunkInShipyard(level, chunkPos.x, chunkPos.z)) || (ship != null && !ShipSettingsKt.getSettings(ship).getShouldGenerateChunks())` just in case something bad happens and someone tries to generate chunks in the shipyard where no ship exists we can prevent it

Reply:
Oh ic i didn't know that there was a difference in isChunkInShipYard and getManagingShip